### PR TITLE
update kernel config to allow audio output  (dtparam=audio=on)

### PR DESCRIPTION
**Pull request by thefinn93**
copied from https://github.com/anisse/beatbox-kernel/commit/f597f35750fa1703a81bef621056225faaec3237, as discussed in https://github.com/gokrazy/gokrazy/discussions/238. I'm not sure if committing vmlinuz or other binaries is desirable. I'm also not sure how to test if this breaks anything else, I'm not super familiar with doing custom kernel builds.

### CODE REVIEW

File: cmd/gokr-build-kernel/build.go [hunk 1]
```diff
@@ -1028,6 +1028,7 @@ CONFIG_SND_BCM2835_SOC_I2S=y
 ##
 CONFIG_SND_SOC_QCOM=y
 CONFIG_SND_SOC_APQ8016_SBC=y
+CONFIG_HIDRAW=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1031]: This line is unrelated to audio. We can also enable hidraw, but that should be a separate pull request.

File: cmd/gokr-build-kernel/build.go [hunk 2]
```diff
@@ -1040,6 +1041,13 @@ CONFIG_SND_SOC_TEGRA_TRIMSLICE=y
 CONFIG_SND_SOC_TEGRA_ALC5632=y
 CONFIG_SND_SOC_TEGRA_MAX98090=y
 CONFIG_SND_SOC_TEGRA_RT5677=y
+CONFIG_STAGING=y
+CONFIG_SND=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1045]: `CONFIG_SND` already seems to be =y in the current kernel config, so this line can be removed

File: cmd/gokr-build-kernel/build.go [hunk 3]
```diff
@@ -1040,6 +1041,13 @@ CONFIG_SND_SOC_TEGRA_TRIMSLICE=y
...
+CONFIG_OF=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1046]: `CONFIG_OF` already seems to be =y in the current kernel config, so this line can be removed

File: cmd/gokr-build-kernel/build.go [hunk 4]
```diff
@@ -1040,6 +1041,13 @@ CONFIG_SND_SOC_TEGRA_TRIMSLICE=y
...
+CONFIG_BCM_VIDEOCORE=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1047]: `CONFIG_BCM_VIDEOCORE` already seems to be =y in the current kernel config, so this line can be removed

File: cmd/gokr-build-kernel/build.go [hunk 5]
```diff
@@ -1040,6 +1041,13 @@ CONFIG_SND_SOC_TEGRA_TRIMSLICE=y
...
+CONFIG_BCM2835_VCHIQ=y
+CONFIG_SND_BCM2835=y
+CONFIG_HW_RANDOM_BCM2835=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1050]: `CONFIG_HW_RANDOM_BCM2835` already seems to be =y in the current kernel config, so this line can be removed

File: cmd/gokr-build-kernel/build.go [hunk 6]
```diff
@@ -1040,6 +1041,13 @@ CONFIG_SND_SOC_TEGRA_TRIMSLICE=y
 CONFIG_SND_SOC_TEGRA_ALC5632=y
 CONFIG_SND_SOC_TEGRA_MAX98090=y
 CONFIG_SND_SOC_TEGRA_RT5677=y
+CONFIG_STAGING=y
+CONFIG_SND=y
+CONFIG_OF=y
+CONFIG_BCM_VIDEOCORE=y
+CONFIG_BCM2835_VCHIQ=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1048]: `CONFIG_BCM2835_VCHIQ` will be auto-selected by `CONFIG_SND_BCM2835`, so this line can be removed

File: cmd/gokr-build-kernel/build.go [hunk 7]
```diff
@@ -1040,6 +1041,13 @@ CONFIG_SND_SOC_TEGRA_TRIMSLICE=y
 CONFIG_SND_SOC_TEGRA_ALC5632=y
 CONFIG_SND_SOC_TEGRA_MAX98090=y
 CONFIG_SND_SOC_TEGRA_RT5677=y
+CONFIG_STAGING=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1044]: `CONFIG_STAGING` already seems to be =y in the current kernel config, so this line can be removed

File: cmd/gokr-build-kernel/build.go [hunk 8]
```diff
@@ -1040,6 +1041,13 @@ CONFIG_SND_SOC_TEGRA_TRIMSLICE=y
...
+CONFIG_SND=y
+CONFIG_OF=y
+CONFIG_BCM_VIDEOCORE=y
+CONFIG_BCM2835_VCHIQ=y
+CONFIG_SND_BCM2835=y
```
Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1049]: Can you move this line to line 1024, into the `file: sound/soc/bcm/Kconfig` block?

Comment (stapelberg) [inline cmd/gokr-build-kernel/build.go:1049]: Can you move this line to line 1024, into the `file: sound/soc/bcm/Kconfig` block?

**Review by stapelberg — CHANGES_REQUESTED** [review]
Yes, updating the kernel binary is needed, so the PR seems correct. 

Regarding breakage, as long as the other kernel options are still selected as before (which seems to be the case), nothing should stop working. We’ll commit and see.

**Comment by thefinn93** [comment]
Cool, I wasn't sure how many of those were actually needed. I have updated the commit so the only config change is the addition of one `CONFIG_SND_BCM2835=y`

**Review by stapelberg — APPROVED** [review]
